tck/src/main/java/org/eclipse/sparkplug/tck/test/host/SessionTerminationTest.java

```diff
@@ -193,6 +194,41 @@ public void disconnect(final @NotNull String clientId, final @NotNull Disconnect
 		}
 	}
 
+	@SpecAssertion(
+			section = Sections.OPERATIONAL_BEHAVIOR_SPARKPLUG_HOST_APPLICATION_SESSION_TERMINATION,
+			id = ID_OPERATIONAL_BEHAVIOR_HOST_APPLICATION_DISCONNECT_INTENTIONAL)
+	@SpecAssertion(
+			section = Sections.OPERATIONAL_BEHAVIOR_SPARKPLUG_HOST_APPLICATION_SESSION_TERMINATION,
+			id = ID_OPERATIONAL_BEHAVIOR_HOST_APPLICATION_TERMINATION)
+	@SpecAssertion(
+			section = Sections.PAYLOADS_DESC_STATE_DEATH,
+			id = ID_HOST_TOPIC_PHID_DEATH_PAYLOAD_TIMESTAMP_DISCONNECT_CLEAN)
+	@Override
+	public void onDisconnect(DisconnectEventInput disconnectEventInput) {
```

Comment:
I deliberately didn't add these checks to the onDisconnect method because this doesn't imply a disconnect packet was sent. 

At least this assertion seems like we shouldn't check off here:

    ID_OPERATIONAL_BEHAVIOR_HOST_APPLICATION_DISCONNECT_INTENTIONAL

"In the case of intentionally disconnecting, an MQTT DISCONNECT packet MUST be sent immediately after the Death message is published."

Reply:
I think the spec isn't worded quite right. I'm going to rework these statements a bit to support both disconnecting with a DISCONNECT and without and adjust the tests accordingly.